app/Actions/Photo/Strategies/RotateStrategy.php

```diff
@@ -0,0 +1,182 @@
+		$oldOriginalSizeVariant = $this->photo->size_variants->getSizeVariant(SizeVariant::ORIGINAL);
+		$oldOriginalFullPath = $oldOriginalSizeVariant->full_path;
+		$oldOriginalWidth = $oldOriginalSizeVariant->width;
+		$oldOriginalHeight = $oldOriginalSizeVariant->height;
+		$oldChecksum = $this->photo->checksum;
+		$oldExtension = Helpers::getExtension($oldOriginalFullPath);
+		$tmpFullPath = Helpers::createTemporaryFile($oldExtension);
+
+		// Rotate the image and save result as the temporary file
+		/** @var ImageHandlerInterface $imageHandler */
+		$imageHandler = resolve(ImageHandlerInterface::class);
+		if ($imageHandler->rotate($oldOriginalFullPath, ($this->direction == 1) ? 90 : -90, $tmpFullPath) === false) {
+			$msg = 'Failed to rotate ' . $oldOriginalFullPath;
+			Logs::error(__METHOD__, __LINE__, $msg);
+			throw new \RuntimeException($msg);
+		}
+
+		// The file size and checksum may have changed after the rotation.
+		/* @var Extractor $metadataExtractor */
+		$metadataExtractor = resolve(Extractor::class);
+		$this->photo->filesize = $metadataExtractor->filesize($tmpFullPath);
+		$this->photo->checksum = $metadataExtractor->checksum($tmpFullPath);
+		$this->photo->save();
+
+		// Delete all size variants from current photo, this will also take
+		// care of erasing the actual "physical" files from storage and any
+		// potential symbolic link which points to one of the original files.
+		// This will bring photo entity into the same state as it would be if
+		// we were importing an new photo.
+		$this->photo->size_variants->delete();
+
+		// Initialize factory for size variants
+		$this->parameters->sourceFileInfo = new SourceFileInfo(
+			pathinfo($oldOriginalFullPath, PATHINFO_BASENAME),
+			$this->photo->type,
+			$tmpFullPath
+		);
+		/** @var SizeVariantNamingStrategy $namingStrategy */
+		$namingStrategy = resolve(SizeVariantNamingStrategy::class);
+		$namingStrategy->setFallbackExtension($this->parameters->sourceFileInfo->getOriginalFileExtension());
+		/** @var SizeVariantFactory $sizeVariantFactory */
+		$sizeVariantFactory = resolve(SizeVariantFactory::class);
+		$sizeVariantFactory->init($this->photo, $namingStrategy);
+
+		// Create size variant for rotated original
+		// Note that his also creates a different file name than before
+		// because the checksum of the photo has changed.
+		// Using a different filename allows to avoid caching effects.
+		// Sic! Swap width and height here, because the image has been rotated
+		$newOriginalSizeVariant = $sizeVariantFactory->createOriginal($oldOriginalHeight, $oldOriginalWidth);
+		$this->putSourceIntoFinalDestination($newOriginalSizeVariant->full_path);
+
+		// Create remaining size variants
+		$newSizeVariants = null;
+		try {
+			$newSizeVariants = $sizeVariantFactory->createSizeVariants();
+			// add new original size variant to collection of newly created
+			// size variant; we need this to correctly update the duplicates
```

Comment:
Typo: variant -> variants (in the 2nd line of the comment)

Reply:
Fixed